AppCenterData/AppCenterData/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -263,4 +367,49 @@ + (NSString *)tableNameForPartition:(NSString *)partition {
   return pendingDocuments;
 }
 
+- (BOOL)hasPendingOperationsForPartition:(NSString *)partition {
+  NSString *tableName = [MSDBDocumentStore tableNameForPartition:partition];
+  NSString *selectionQuery =
```

Comment:
How about doing a select count(*) instead of trying to get all documents and instantiating them? It would be much cheaper (perf wise).